### PR TITLE
bpo-36026: make descr error message consistent

### PR DESCRIPTION
**Pull request by methane**
`set.add(0)` and `set.add.__get__(0)` now raise TypeError with same error message.

https://bugs.python.org/issue36026

### CODE REVIEW

File: Objects/descrobject.c [hunk 1]
```diff
@@ -78,8 +78,8 @@ descr_check(PyDescrObject *descr, PyObject *obj, PyObject **pres)
     }
     if (!PyObject_TypeCheck(obj, descr->d_type)) {
         PyErr_Format(PyExc_TypeError,
-                     "descriptor '%V' for '%s' objects "
```
Comment (serhiy-storchaka) [inline Objects/descrobject.c:81]: Why has this been changed?

There was a difference with classmethod descriptors, but classmethod descriptors were special, they are applied to types, not instances.

**Review by serhiy-storchaka — COMMENTED** [review]
Could you please provide tests?

File: Objects/descrobject.c (hunk 1 above)
Comment (methane) [reply Objects/descrobject.c:81]: This message is for method, not classmethod.  But I changed classmethod error message too.
I feel same error message makes sense for classmethod too.
Discuss about actual error message on bpo.

File: Objects/descrobject.c [hunk 2]
```diff
@@ -314,21 +312,20 @@ classmethoddescr_call(PyMethodDescrObject *descr, PyObject *args,
...
         PyErr_Format(PyExc_TypeError,
-                     "descriptor '%V' "
-                     "requires a subtype of '%.100s' "
-                     "but received '%.100s",
+                     "descriptor '%V' for type '%.100s' "
+                     "doesn't apply to a '%.100s' type",
                      descr_name((PyDescrObject *)descr), "?",
                      PyDescr_TYPE(descr)->tp_name,
-                     self->ob_type->tp_name);
+                     ((PyTypeObject*)self)->tp_name);
```
Comment (methane) [inline Objects/descrobject.c]: Maybe, this is a really old bug ([commit](https://github.com/python/cpython/commit/7295c6a871f9cc42a4209a8eebe2e0974194a2a3)).

```
>>> desc = set.add
>>> desc(int)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'add' requires a 'set' object but received a 'type'
```

This should be `...but received a 'int'`, not 'type'.

Comment (serhiy-storchaka) [reply Objects/descrobject.c]:
```
>>> desc(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'add' requires a 'set' object but received a 'int'
>>> desc(set)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'add' requires a 'set' object but received a 'type'
```

Comment (serhiy-storchaka) [reply Objects/descrobject.c]: This does not look like a bug to me. If you will change it to produce `...but received a 'int'`, you will get `... requires a 'set' object but received a 'set'` for `desc(set)`, and I have no idea what you will get for `descr(1)`.

Comment (methane) [reply Objects/descrobject.c]: I'm sorry, my example was wrong.  This is classmethoddesc_call.

```
>>> desc2 = dict.__dict__['fromkeys']
>>> desc2(int, [])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'fromkeys' requires a subtype of 'dict' but received 'type
```

`'type` must be `'int'`.

Comment (methane) [reply Objects/descrobject.c]: filed https://bugs.python.org/issue36433

**Pull request merged**